src/sql/workbench/browser/modelComponents/text.component.ts

```diff
@@ -18,7 +18,13 @@ import { SafeHtml, DomSanitizer } from '@angular/platform-browser';
 @Component({
 	selector: 'modelview-text',
 	template: `
-		<p [style.width]="getWidth()" [innerHTML]="getValue()" [ngStyle]="this.CSSStyles" (click)="onClick()"></p>`
+		<div style="display:flex;flex-flow:row;align-items:center;" [style.width]="getWidth()">
+			<p [innerHTML]="getValue()" [ngStyle]="this.CSSStyles"></p>
+			<p  *ngIf="requiredIndicator" style="color:red;margin-left:5px;">*</p>
+			<div *ngIf="description" tabindex="0" class="modelview-text-tooltip" [attr.aria-label]="description">
+				<div class="modelview-text-tooltip-content" [innerHTML]="description"></div>
+			</div>
+		<div>`
```

Comment:
Let's be really careful here...this is a lot of changes for a very commonly used component. Have you look at other places this is used to make sure they still look OK. Please make sure you look at the BDC dashboard for sure.